code/game/objects/items/weapons/material/swords.dm

```diff
@@ -149,4 +149,61 @@
 	icon_state = "amohdan_sword"
 	item_state = "amohdan_sword"
 	contained_sprite = 1
-	slot_flags = SLOT_BELT
\ No newline at end of file
+	slot_flags = SLOT_BELT
+
+// improvised sword
+/obj/item/weapon/material/sword/improvised_sword
+	name = "selfmade sword"
+	desc = "A crudely made, rough looking sword. Still appears to be quite deadly."
+	use_material_name = FALSE
+	icon = 'icons/obj/improvised.dmi'
+	icon_state = "sword"
+	item_state = "sword"
+	force_divisor = 0.3 // 18 damage
+	contained_sprite = 1
+	slot_flags = SLOT_BELT
+
+// the things needed to create the above
+/obj/item/weapon/material/sword_hilt
+	name = "hilt"
+	desc = "A hilt without a blade, quite useless."
+	use_material_name = FALSE
+	icon = 'icons/obj/improvised.dmi'
+	icon_state = "hilt"
+	unbreakable = TRUE
+	force_divisor = 0.05
+	thrown_force_divisor = 0.2
+
+/obj/item/weapon/material/sword_hilt/attackby(var/obj/O, mob/user)
+	if(istype(O, /obj/item/weapon/material/sword_blade))
+		var/obj/item/weapon/material/sword_blade/blade = O
+		var/obj/item/weapon/material/sword/improvised_sword/new_sword = new(src.loc, src.material.name)

```

Comment:
```suggestion
		var/obj/item/weapon/material/sword/improvised_sword/new_sword = new(src.loc, blade.material.name)
```
This makes it so that it uses the material of the blade always, so a plasteel sword with a wooden hilt won't give papercuts.
I'd recommend refactoring it so that its integrity depends on the hilt and its damage depends on its blade.